### ISSUE TITLE
Git clone problem

### ISSUE THREAD

**Issue by leanfrancucci** [Microsoft/WSL#3180]
- Your Windows build number:  Microsoft Windows [Version 10.0.16299.371]

- What you're doing and what's happening:  
1. From the last year, I manage a several git repositories using WSL, containing Visual Studio projects. 
2. When I clone any of them using WSL I cannot compile the VS projects anymore. 
3. However, if I copy the whole project directory (previously cloned repo) to an any other directory only using Win's Explorer I can succesfully compile it. 
4. If I copy the previously cloned repo using WSL, the compiling fails.
5. I launch VS from Windows.
6. As additional data, the compiler error is: "fatal error C1033: cannot open program database 'c:\project\path\vc110.pdb"

- What's wrong / what should be happening instead:
I would expect that the VS projects contained in a git repo and managed from WSL can be normally used. This issue arose from the current Windows 10 version. I suspect it is a file permissions issue.

Please, could somebody help me?.

**Comment by Biswa96** [comment]
Run these commands in WSL:
```
git config --global core.autocrlf input
git config --global core.symlinks true
```

Set up your username and password:

```
git config --global user.name "User Name"
git config --global user.email "user.name@mail.com"
```

**Comment by GPhilo** [comment]
I ran into a very similar problem (and opened a SO question [here](https://stackoverflow.com/q/50260328/3214872)), although in my case it was the compiler not finding existing files and intellisense being completely broken. It also appears MSBuild can't find its tracking log files. Adding the suggested configuration and repulling the repository doesn't seem to fix the error. I didn't have time yet to check if cloning on a fresh folder still shows the error.
From what I could see, though, whatever error is there seems not affecting Explorer or other Windows features (I could navigate the folders, open the files, copy them around without problems).

**Comment by leanfrancucci** [comment]
@Biswa96 I have already set up the options that you suggest me but the issue remains.
@GPhilo I think that we deal with same problem.

**Comment by therealkenc** [comment]
> I cannot compile the VS projects anymore

"Anymore" relative to what? This needs repro steps to get any kind of help. One way to do that would be to put a toy Hello World repo on Github that exhibits the behaviour, and the steps with clean install including real paths to reproduce the problem.

But on a lark, try remounting `/mnt/c` thus:
```
$ sudo umount /mnt/c
$ sudo mount -t drvfs -o metadata,uid=1000,gid=1000,case=off C: /mnt/c
```
Use a different `uid/gid` if `1000` isn't you.

**Comment by nsubtil** [comment]
I'm seeing similar issues after upgrading to RS4: cloning GLFW in WSL and running the build outside WSL causes the compiler to be unable to find the library's header files during the build.

Cloning the same repo outside WSL makes the problem go away, and so does reverting back to RS3.

VSCode with vscode-cmake-extensions installed will also have trouble finding the root CMakeLists.txt during the build.

It seems like remounting with the options that @therealkenc suggested makes at least the build issues go away. Is there some guidance on whether this is the right thing to do?

**Comment by GPhilo** [comment]
@therealkenc I tried making a toy repo that exposes the problem yesterday, but I haven't been able to reproduce the bug yet.

One more odd thing though: the repository I cloned has multiple projects. Only the one using Qt cannot compile, because some generated files that are put in a subdirectory can't be found from the compiler. The other projects, on the other hand, can build.
Still, on all projects MSBuild is broken (it always rebuilds everything because it can't find the build log) and intellisense doesn't work at all (can't find symbols, suggestions, anything)
